src/shogun-core-main/src/main/java/de/terrestris/shoguncore/util/interceptor/secure/WfsRequestInterceptor.java

```diff
@@ -0,0 +1,247 @@
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Interceptor class for WFS requests. Adds basic auth headers based on the GS
+ * properties by default and checks for permission on layer
+ *
+ * @author Nils Bühner
+ * @author Johannes Weskamm
+ *
+ */
+public class WfsRequestInterceptor implements WfsRequestInterceptorInterface {
+
+    /**
+     *
+     */
+    private URI appUri = null;
+
+    /**
+     *
+     */
+    private static final Logger LOG = getLogger(WfsRequestInterceptor.class);
+
+    /**
+     *
+     */
+    @Autowired
+    @Qualifier("userService")
+    protected UserService<User, UserDao<User>> userService;
+
+    /**
+     *
+     */
+    @Autowired
+    @Qualifier("layerService")
+    protected LayerService<Layer, LayerDao<Layer>> layerService;
+
+    /**
+     *
+     */
+    @Value("${geoserver.username:}")
+    private String gsUser;
+
+    /**
+     *
+     */
+    @Value("${geoserver.password:}")
+    private String gsPass;
+
+   /**
+     *
+     */
+    @Override
+    public MutableHttpServletRequest interceptGetCapabilities(MutableHttpServletRequest request) {
+        LOG.debug("Intercepting WFS GetCapabilities");
+        return new GeoserverAuthHeaderRequest(request, gsUser, gsPass);
+    }
+
+    /**
+     *
+     */
+    @Override
+    public MutableHttpServletRequest interceptDescribeFeatureType(MutableHttpServletRequest request) {
+        LOG.debug("Intercepting WFS DescribeFeatureType");
+        return new GeoserverAuthHeaderRequest(request, gsUser, gsPass);
+    }
+
+    /**
+     *
+     */
+    @Override
+    public MutableHttpServletRequest interceptGetFeature(MutableHttpServletRequest request) {
+        LOG.debug("Intercepting WFS GetFeature");
+        return isAllowed(request, "READ") ? new GeoserverAuthHeaderRequest(
+            request, gsUser, gsPass) : forbidRequest(request);
+    }
+
+    /**
+     *
+     */
+    @Override
+    public MutableHttpServletRequest interceptLockFeature(MutableHttpServletRequest request) {
+        LOG.debug("Intercepting WFS LockFeature");
+        return isAllowed(request, "UPDATE") ? new GeoserverAuthHeaderRequest(
+            request, gsUser, gsPass) : forbidRequest(request);
+    }
+
+    /**
+     *
+     */
+    @Override
+    public MutableHttpServletRequest interceptTransaction(MutableHttpServletRequest request) {
+        LOG.debug("Intercepting WFS Transaction");
+        return isAllowed(request, "UPDATE") ? new GeoserverAuthHeaderRequest(
+            request, gsUser, gsPass) : forbidRequest(request);
+    }
+
+    /**
+     * Check for permission on layer by getting it through service.
+     * For update permissions, we check through custom method.
+     *
+     * @param request   The request
+     * @param paramName The optional parameter name for the layer parameter
+     * @param method    The access method like read or write
+     * @return if the layer is allowed to read for current user
+     */
+    private boolean isAllowed(
```

Comment:
Did you check if it's possible to reuse the `hasPermission` method of the PermissionEvaluator?

Reply:
yes and i did not find a way to do so :-/

Reply:
Too bad.